src/libraries/System.Private.CoreLib/src/System/Int32.cs

```diff
@@ -504,37 +504,27 @@ int IBinaryInteger<int>.GetShortestBitLength()
         /// <inheritdoc cref="IBinaryInteger{TSelf}.TryWriteBigEndian(Span{byte}, out int)" />
         bool IBinaryInteger<int>.TryWriteBigEndian(Span<byte> destination, out int bytesWritten)
         {
-            if (destination.Length >= sizeof(int))
+            if (BinaryPrimitives.TryWriteInt32BigEndian(destination, m_value))
```

Comment:
If these were instead:
```C#
bool result = BinaryPrimitives.TryWriteInt32BigEndian(destination, m_value);
bytesWritten = result ? sizeof(int) : 0;
return result;
```
would the JIT be able to eliminate the branch and just use a conditional move? Or does it already do that?